operator/controllers/lifecycle/keptnappcreationrequest/controller.go

```diff
@@ -45,13 +70,169 @@ type KeptnAppCreationRequestReconciler struct {
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile
+//
+//nolint:gocyclo
 func (r *KeptnAppCreationRequestReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	creationRequest := &lifecycle.KeptnAppCreationRequest{}
+
+	if err := r.Get(ctx, req.NamespacedName, creationRequest); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("could not retrieve KeptnAppCreationRequest: %w", err)
+	}
+
+	// check if we already have an app that has not been created by this controller
+
+	appFound := false
+	keptnApp := &lifecycle.KeptnApp{}
+	name := req.NamespacedName
+	name.Name = creationRequest.Spec.AppName
+	if err := r.Get(ctx, name, keptnApp); err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("No KeptnApp found for KeptnAppCreationRequest", "KeptnAppCreationRequest", creationRequest)
+		} else {
+			return ctrl.Result{}, fmt.Errorf("could not retrieve KeptnApp %w", err)
+		}
+	} else {
+		appFound = true
+	}
+
+	// if the found app has not been created by this controller, we are done at this point - we don't want to mess with what the user has created
+	if appFound && keptnApp.Labels[common.K8sRecommendedManagedByAnnotations] != managedByKLT {
+		r.Log.Info("User defined KeptnApp found for KeptnAppCreationRequest", "KeptnAppCreationRequest", creationRequest)
+		if err := r.Delete(ctx, creationRequest); err != nil {
+			r.Log.Error(err, "Could not delete KeptnAppCreationRequest", "KeptnAppCreationRequest", creationRequest)
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// check if discovery deadline has expired or if the application is a single service app
+	if !r.shouldCreateApp(creationRequest) {
+		r.Log.Info("Discovery deadline not expired yet", "KeptnAppCreationRequest", creationRequest)
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
I would make a small perf improvement here and requeue it after `deadline - now`